internal/datastore/crdb/readwrite.go

```diff
@@ -0,0 +1,285 @@
+		}
+	}
+
+	return nil
+}
+
+func exactRelationshipClause(r *v1.Relationship) sq.Eq {
+	return sq.Eq{
+		colNamespace:        r.Resource.ObjectType,
+		colObjectID:         r.Resource.ObjectId,
+		colRelation:         r.Relation,
+		colUsersetNamespace: r.Subject.Object.ObjectType,
+		colUsersetObjectID:  r.Subject.Object.ObjectId,
+		colUsersetRelation:  stringz.DefaultEmpty(r.Subject.OptionalRelation, datastore.Ellipsis),
+	}
+}
+
+func (rwt *crdbReadWriteTXN) DeleteRelationships(filter *v1.RelationshipFilter) error {
+	ctx, span := tracer.Start(datastore.SeparateContextWithTracing(rwt.ctx), "DeleteRelationships")
+	defer span.End()
+	// Add clauses for the ResourceFilter
+	query := queryDeleteTuples.Where(sq.Eq{colNamespace: filter.ResourceType})
+	tracerAttributes := []attribute.KeyValue{common.ObjNamespaceNameKey.String(filter.ResourceType)}
+	if filter.OptionalResourceId != "" {
+		query = query.Where(sq.Eq{colObjectID: filter.OptionalResourceId})
+		tracerAttributes = append(tracerAttributes, common.ObjIDKey.String(filter.OptionalResourceId))
+	}
+	if filter.OptionalRelation != "" {
+		query = query.Where(sq.Eq{colRelation: filter.OptionalRelation})
+		tracerAttributes = append(tracerAttributes, common.ObjRelationNameKey.String(filter.OptionalRelation))
+	}
+	rwt.addOverlapKey(filter.ResourceType)
+
+	// Add clauses for the SubjectFilter
+	if subjectFilter := filter.OptionalSubjectFilter; subjectFilter != nil {
+		query = query.Where(sq.Eq{colUsersetNamespace: subjectFilter.SubjectType})
+		tracerAttributes = append(tracerAttributes, common.SubNamespaceNameKey.String(subjectFilter.SubjectType))
+		if subjectFilter.OptionalSubjectId != "" {
+			query = query.Where(sq.Eq{colUsersetObjectID: subjectFilter.OptionalSubjectId})
+			tracerAttributes = append(tracerAttributes, common.SubObjectIDKey.String(subjectFilter.OptionalSubjectId))
+		}
+		if relationFilter := subjectFilter.OptionalRelation; relationFilter != nil {
+			query = query.Where(sq.Eq{colUsersetRelation: stringz.DefaultEmpty(relationFilter.Relation, datastore.Ellipsis)})
+			tracerAttributes = append(tracerAttributes, common.SubRelationNameKey.String(relationFilter.Relation))
+		}
+		rwt.addOverlapKey(subjectFilter.SubjectType)
+	}
+	span.SetAttributes(tracerAttributes...)
+	sql, args, err := query.ToSql()
+	if err != nil {
+		return fmt.Errorf(errUnableToDeleteRelationships, err)
+	}
+
+	modified, err := rwt.tx.Exec(ctx, sql, args...)
+	if err != nil {
+		return fmt.Errorf(errUnableToDeleteRelationships, err)
+	}
+
+	rwt.relCountChange -= modified.RowsAffected()
+
+	return nil
+}
+
+func (rwt *crdbReadWriteTXN) WriteNamespaces(newConfigs ...*core.NamespaceDefinition) error {
+	query := queryWriteNamespace
+
+	for _, newConfig := range newConfigs {
+		rwt.addOverlapKey(newConfig.Name)
+
+		serialized, err := proto.Marshal(newConfig)
+		if err != nil {
+			return fmt.Errorf(errUnableToWriteConfig, err)
+		}
+		query = query.Values(newConfig.Name, serialized)
```

Comment:
We might have a risk of hitting query limits here too

Reply:
Yeah, there is a hard upper bound on how big our schema can get now, but it's unlikely to hit this limit without first hitting the overall schema limit.